srai/loaders/osm_loaders/osm_tile_loader.py

```diff
@@ -0,0 +1,127 @@
+                to request. Defaults to None.
+            collector_factory (Optional[Union[str, Callable[[], DataCollector]]], optional):Function
+                returning DataCollector. If None uses InMemoryDataCollector. Defaults to None.
+                    If `return` uses  InMemoryDataCollector
+                    If `save` uses  SavingDataCollector
+            storage_path (Optional[Union[str, Path]], optional): path to save data,
+                used with SavingDataCollector. Defaults to None.
+
+        References:
+        1. https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames
+        """
+        self.zoom = zoom
+        self.verbose = verbose
+        self.resource_type = resource_type
+        self.base_url = urljoin(tile_server_url, "{0}/{1}/{2}." + resource_type)
+        self.auth_token = auth_token
+        self.save_path = storage_path
+        self.collector_factory = (
+            self._get_collector_factory(collector_factory)
+            if collector_factory is not None
+            else lambda: InMemoryDataCollector()
+        )
+        self.regionizer = SlippyMapRegionizer(z=self.zoom)
+
+    def _get_collector_factory(
+        self, storage_strategy: Union[str, Callable[[], DataCollector]]
+    ) -> Callable[[], DataCollector]:
+        if isinstance(storage_strategy, str):
+            if storage_strategy == "save" and self.save_path is None:
+                raise ValueError
+            elif self.save_path is not None:
+                save_path: Union[Path, str] = self.save_path
+            return {
+                "save": lambda: SavingDataCollector(save_path, f_extension=self.resource_type),
+                "return": lambda: InMemoryDataCollector(),
+            }[storage_strategy]
+        else:
+            return storage_strategy
+
+    def get_tile_by_x_y(self, x: int, y: int) -> Image.Image:
+        """
+        Downloads single tile from tile server.
+
+        Args:
+            x: x tile x coordinate
+            y: y tile y coordinate
+        """
+        url = self.base_url.format(self.zoom, x, y)
+        if self.verbose:
+            print(f"Getting tile from url: {url}")
+        content = requests.get(url, params={"access_token": f"{self.auth_token}"}).content
+        return Image.open(BytesIO(content))
+
+    def get_tile_by_region_name(self, name: str) -> pd.DataFrame:
+        """
+        Returns all tiles of region.
+
+        Args:
+            name: area name, as in geocode_to_region_gdf
+            return_rect: if true returns tiles out of area to keep rectangle shape of img of joined
+                tiles.
+        """
+        tiles_collector = self.collector_factory()
+        gdf = geocode_to_region_gdf(name)
+        regions = self.regionizer.transform(gdf=gdf)
+        data_series = regions.apply(
+            lambda row: self._get_tile_for_area(row, tiles_collector), axis=1
+        )
+        return pd.DataFrame(data_series, columns=["tile"])
```

Comment:
Why cannot I pass geodataframe myself? Passing only name takes away functionality of `geocode_to_region_gdf` (such as passing by osmid). I suggest ditching passing by name and only implement passing gdf

Reply:
We try to be consistent with function names across loaders. I would suggest renaming this function to `load` as in other loaders. If it comes to function signature and flow of data we try to pass GeoDataFrames (or just DataFrames) always

Reply:
I really like this idea. 

Why don't put some interface or abstract class, like Regionizer in regionizers, to show explicitly you want to be consistent?